src/software/simulation/physics/physics_ball_test.cpp

```diff
@@ -256,80 +268,99 @@ TEST(PhysicsBallTest, test_apply_force_to_change_direction_of_moving_ball)
     EXPECT_LT((ball.velocity() - Vector(1, 2)).length(), 0.05);
 }
 
 TEST(PhysicsBallTest, test_apply_impulse_to_stop_moving_ball)
 {
     b2Vec2 gravity(0, 0);
     auto world = std::make_shared<b2World>(gravity);
-    Ball ball_parameter(Point(0, 0), Vector(2, -1), Timestamp::fromSeconds(0));
-    auto physics_ball = PhysicsBall(world, ball_parameter, 1.0, 9.8);
+    BallState initial_ball_state(Point(0, 0), Vector(2, -1));
+    auto physics_ball = PhysicsBall(world, initial_ball_state, 1.0, 9.8);
 
     physics_ball.applyImpulse(Vector(-2, 1));
     // 5 and 8 here are somewhat arbitrary values for the velocity and position
     // iterations but are the recommended defaults from
     // https://www.iforce2d.net/b2dtut/worlds
     world->Step(1.0 / 60.0, 5, 8);
-    auto ball = physics_ball.getBallWithTimestamp(Timestamp::fromSeconds(0));
+
+    auto ball = physics_ball.getBallState();
     EXPECT_LT((ball.velocity() - Vector(0, 0)).length(), 0.05);
 }
 
 TEST(PhysicsBallTest, test_apply_impulse_to_change_direction_of_moving_ball)
 {
     b2Vec2 gravity(0, 0);
     auto world = std::make_shared<b2World>(gravity);
-    Ball ball_parameter(Point(0, 0), Vector(-3, -0.5), Timestamp::fromSeconds(0));
-    auto physics_ball = PhysicsBall(world, ball_parameter, 1.0, 9.8);
+    BallState initial_ball_state(Point(0, 0), Vector(-3, -0.5));
+    auto physics_ball = PhysicsBall(world, initial_ball_state, 1.0, 9.8);
 
     physics_ball.applyImpulse(Vector(2, 0.5));
     // 5 and 8 here are somewhat arbitrary values for the velocity and position
     // iterations but are the recommended defaults from
     // https://www.iforce2d.net/b2dtut/worlds
     world->Step(1.0 / 60.0, 5, 8);
-    auto ball = physics_ball.getBallWithTimestamp(Timestamp::fromSeconds(0));
+
+    auto ball = physics_ball.getBallState();
     EXPECT_LT((ball.velocity() - Vector(-1, 0)).length(), 0.05);
 }
 
-TEST(PhysicsBallTest, test_kick_ball)
+TEST(PhysicsBallTest, test_kick_stationary_ball)
 {
     b2Vec2 gravity(0, 0);
     auto world = std::make_shared<b2World>(gravity);
-    Ball ball_parameter(Point(0, 0), Vector(0, 0), Timestamp::fromSeconds(0));
-    auto physics_ball = PhysicsBall(world, ball_parameter, 0.44, 9.8);
+    BallState initial_ball_state(Point(0, 0), Vector(0, 0));
+    auto physics_ball = PhysicsBall(world, initial_ball_state, 0.44, 9.8);
 
     physics_ball.kick(Vector(-2, 3));
     // 5 and 8 here are somewhat arbitrary values for the velocity and position
     // iterations but are the recommended defaults from
     // https://www.iforce2d.net/b2dtut/worlds
     world->Step(1.0 / 60.0, 5, 8);
-    auto ball = physics_ball.getBallWithTimestamp(Timestamp::fromSeconds(0));
+    auto ball = physics_ball.getBallState();
     EXPECT_LT((ball.velocity() - Vector(-2, 3)).length(), 0.05);
 }
 
-TEST(PhysicsBallTest, test_chip_ball_without_collisions)
+TEST(PhysicsBallTest, test_kick_moving_ball)
+{
+    b2Vec2 gravity(0, 0);
```

Comment:
does this mean there's no gravity?

Reply:
correct, but this is gravity in the x-y direction as we think of it (because this is a 2D sim, which we "look down" on). Eg. gravity of (1, 0) would cause the ball to accelerate in the positive x direction.

We have separate gravity and gravity-related calculations for chipping to "mock" 3d space.